### PR TITLE
Update API documentation link on README

### PR DESCRIPTION
**Pull request by jrwrigh**

### CODE REVIEW

**Comment by jeremylt** [comment]
Good catch; thank you!

**Comment by jrwrigh** [comment]
There's another broken link on line 59. I was hoping to put it as a code comment on the PR (since I'm not sure where it should point to), but for whatever reason GitHub won't let me make comments on that line.

File: README.rst [hunk 1]
```diff
@@ -56,7 +56,7 @@ The goal of libCEED is to propose such a format, as well as supporting
 implementations and data structures, that enable efficient operator evaluation
 on a variety of computational device types (CPUs, GPUs, etc.). This new operator
 description is based on algebraically
-`factored form <https://libceed.readthedocs.io/en/latest/libCEEDapi.html>`_,
+`factored form <https://libceed.readthedocs.io/en/latest/api.html>`_,
```
Comment (jedbrown) [inline README.rst:59]: This link is 404, but https://libceed.readthedocs.io/en/latest/libCEEDapi/#finite-element-operator-decomposition is probably more relevant than what it was intending to point to (https://libceed.readthedocs.io/en/latest/api/).

Comment (jrwrigh) [reply README.rst:59]: Yeah, I tried to make a review/comment asking if the link should point to somewhere else, but apparently didn't hit submit on it (TIL you have to hit "Start Review" *and* "Submit Review"). I'll open a new PR for the update.

**Pull request merged**